### PR TITLE
[RFC]  Refactor: remove mb_ptr2len_len, mb_ptr2cells and mb_ptr2cells_len

### PR DESCRIPTION
**Pull request by ZviRackover**
Towards completion of #7401

### CODE REVIEW

**Comment by justinmk** [comment]
Thanks! The tests seem to have caught something:

```
[1m[31m[  ERROR   ][0m[0m [36m...vis/build/neovim/neovim/test/functional/ui/fold_spec.lua[0m @ [36m29[0m: [1mfolded lines works with multibyte text[0m
test/functional/ui/screen.lua:417: Row 1 did not match.
Expected:
  |*{5:^+--  2 lines: Ã¥ è¯­ xÌŽÍ‚Í€Ì‚Í›Í› ïºŽï» ï»‹ÙŽïº®ÙŽïº’Ùï»³ÙŽÙ‘ïº“Â·Â·Â·Â·Â·Â·Â·Â·Â·Â·Â·Â·Â·Â·Â·Â·Â·}|
  |{1:~                                            }|
  |{1:~                                            }|
  |{1:~                                            }|
  |{1:~                                            }|
  |{1:~                                            }|
  |{1:~                                            }|
  |                                             |
Actual:
  |*{5:^+--  2 lines: Ã¥ è¯­ xÌŽÍ‚Í€Ì‚Í›Í›ÌŽÍ‚Í€Ì‚Í›Í›Ì¾Í‚Í€Ì‚Í›Í›Ì¾Ì£Í€Ì‚Í›Í›Ì¾Ì£Ì˜Ì‚Í›Í›Ì¾Ì£Ì˜Ì«Í›Í›Ì¾Ì£Ì˜Ì«Ì²Í›Ì¾Ì£Ì˜Ì«Ì²ÍšÌ¾Ì£Ì˜Ì«Ì²ÍšÌ¨Ì£Ì˜Ì«Ì²ÍšÌ¨ÍŽÌ˜Ì«Ì²ÍšÌ¨ÍŽÍ¢Ì«Ì²ÍšÌ¨ÍŽÍ¢ÍŸÌ²ÍšÌ¨ÍŽÍ¢ÍŸÍšÌ¨ÍŽÍ¢ÍŸÌ¨ÍŽÍ¢ÍŸÍŽÍ¢ÍŸÍ¢ÍŸÍŸ ïºŽï» ï»‹ÙŽÙŽïº­ÙŽÙŽïº‘ÙÙ}|
  |{1:~                                            }|
  |{1:~                                            }|
  |{1:~                                            }|
  |{1:~                                            }|
  |{1:~                                            }|
  |{1:~                                            }|
  |                                             |

To print the expect() call that would assert the current screen state, use
screen:snapshot_util(). In case of non-deterministic failures, use
screen:redraw_debug() to show all intermediate screen states.  

stack traceback:
	test/functional/ui/screen.lua:417: in function 'wait'
	test/functional/ui/screen.lua:288: in function 'expect'
	...vis/build/neovim/neovim/test/functional/ui/fold_spec.lua:48: in function <...vis/build/neovim/neovim/test/functional/ui/fold_spec.lua:29>

[1m[31m[  ERROR   ][0m[0m [36m...s/build/neovim/neovim/test/functional/ui/output_spec.lua[0m @ [36m207[0m: [1mshell command :!  handles multibyte sequences split over buffer boundaries[0m
test/functional/ui/screen.lua:417: Row 4 did not match.
Expected:
  |                                                     |
  |{4:                                                     }|
  |:!./shell-test UTF-8                                 |
  |*Ã¥                                                    |
  |ref: Ã¥Ì²                                               |
  |1: Ã¥Ì²                                                 |
  |2: Ã¥ Ì²                                               |
  |3: Ã¥ Ì²                                               |
  |                                                     |
  |{3:Press ENTER or type command to continue}^              |
Actual:
  |                                                     |
  |{4:                                                     }|
  |:!./shell-test UTF-8                                 |
  |*{2:<c3>^@<a5>}                                           |
  |ref: Ã¥ Ì²                                             |
  |1: {2:<c3>} {2:<a5>} Ì²                                       |
  |2: Ã¥ Ì²                                               |
  |3: Ã¥ {2:<b2>}                                            |
  |                                                     |
  |{3:Press ENTER or type command to continue}^              |

To print the expect() call that would assert the current screen state, use
screen:snapshot_util(). In case of non-deterministic failures, use
screen:redraw_debug() to show all intermediate screen states.  

stack traceback:
	test/functional/ui/screen.lua:417: in function 'wait'
	test/functional/ui/screen.lua:288: in function 'expect'
	...s/build/neovim/neovim/test/functional/ui/output_spec.lua:217: in function <...s/build/neovim/neovim/test/functional/ui/output_spec.lua:207>
```

File: src/nvim/os/shell.c [hunk 1]
```diff
@@ -466,7 +466,7 @@ static void out_data_append_to_screen(char *output, size_t *count,
       //    incomplete UTF-8 sequence that could be composing with the last
       //    complete sequence.
       // This will be corrected when we switch to vterm based implementation
-      int i = *p ? mb_ptr2len_len((char_u *)p, (int)(end-p)) : 1;
+      int i = *p ? utf_ptr2len_len((char_u *)p, (int)(end-p)) : 1;
```
Comment (bfredl) [inline src/nvim/os/shell.c]: `utfc_ptr2len_len` (and similar for all `mb_ptr2len_len` replacements)

**Comment by ZviRackover** [comment]
@justinmk  @bfredl  thanks for pointing out the problems. The latest revision should fix the wrong substitutions.

**Comment by bfredl** [comment]
Remaining lint error:
```
src/nvim/keymap.c:901:  Should leave no spaces after a cast: u'(int) ('  [whitespace/cast] [2]
```
otherwise LGTM.

**Comment by ZviRackover** [comment]
Latest revision should fix the lint error

**Pull request merged**